Emby.Server.Implementations/Playlists/PlaylistManager.cs

```diff
@@ -189,41 +189,45 @@ public void AddToPlaylist(string playlistId, IEnumerable<Guid> itemIds, Guid use
 
         private void AddToPlaylistInternal(string playlistId, IEnumerable<Guid> itemIds, User user, DtoOptions options)
         {
-            var playlist = _libraryManager.GetItemById(playlistId) as Playlist;
-
-            if (playlist == null)
-            {
-                throw new ArgumentException("No Playlist exists with the supplied Id");
-            }
-
-            var list = new List<LinkedChild>();
+            // Retrieve the existing playlist
+            var playlist = _libraryManager.GetItemById(playlistId) as Playlist
+                ?? throw new ArgumentException("No Playlist exists with Id " + playlistId);
 
+            // Retrieve all the items to be added to the playlist
             var items = GetPlaylistItems(itemIds, playlist.MediaType, user, options)
                 .Where(i => i.SupportsAddingToPlaylist)
                 .ToList();
```

Comment:
I think we can omit `.ToList()` here as we're probably going over it only once down below

Reply:
It's iterated twice, `items.Count` is called below to check if any duplicates were removed